pages/api/babydegen-metrics.js

```diff
@@ -143,73 +188,97 @@ const fetchOptimusMetrics = async () => {
   }
 
   const populationMetrics = populationResult.value;
-  const defaultMetrics = {
-    latestUsdcApr: null,
-    latestEthApr: null,
-    stakingAprCalculated: null,
-  };
-
-  if (!populationMetrics || populationMetrics.length === 0) {
-    return defaultMetrics;
+  const olasUsdPrice =
+    olasUsdPriceResult.status === 'fulfilled' ? olasUsdPriceResult.value : 0;
+  if (olasUsdPriceResult.status !== 'fulfilled') {
+    console.error('OLAS USD price fetch failed:', olasUsdPriceResult.reason);
   }
 
-  const latest = populationMetrics[populationMetrics.length - 1];
-  const baseMetrics = {
-    latestUsdcApr: toNumber(latest?.sma7dProjectedUnrealisedPnL),
-    latestEthApr: toNumber(latest?.sma7dEthAdjustedProjectedUnrealisedPnL),
-    stakingAprCalculated: null,
-  };
-
+  const stakingSnapshots =
+    stakingResult.status === 'fulfilled' ? stakingResult.value : null;
   if (stakingResult.status !== 'fulfilled') {
     console.error(
       'Optimism staking snapshots fetch failed:',
       stakingResult.reason,
     );
-    return baseMetrics;
   }
 
-  const stakingSnapshots = stakingResult.value;
-  if (!stakingSnapshots || stakingSnapshots.length === 0) {
-    return baseMetrics;
+  const metrics = buildAprMetrics({
+    populationMetrics,
+    stakingSnapshots,
+    olasUsdPrice,
+  });
+
+  return metrics ?? { ...EMPTY_APR_METRICS };
+};
+
+const fetchModiusMetrics = async () => {
+  const [populationResult, stakingResult] = await Promise.allSettled([
+    fetchModiusPopulationMetrics(),
+    fetchModeStakingSnapshots(),
+  ]);
+
+  if (populationResult.status !== 'fulfilled') {
+    console.error(
+      'Modius population metrics fetch failed:',
+      populationResult.reason,
+    );
+    return null;
   }
 
-  const olasUsdPrice =
-    olasUsdPriceResult.status === 'fulfilled' ? olasUsdPriceResult.value : 0;
-  if (olasUsdPriceResult.status !== 'fulfilled') {
-    console.error('OLAS USD price fetch failed:', olasUsdPriceResult.reason);
+  const populationMetrics = populationResult.value;
+  const olasUsdPrice = MODIUS_FIXED_OLAS_PRICE_USD;
+
+  const stakingSnapshots =
+    stakingResult.status === 'fulfilled' ? stakingResult.value : null;
+  if (stakingResult.status !== 'fulfilled') {
+    console.error('Mode staking snapshots fetch failed:', stakingResult.reason);
   }
 
-  const stakingApr = calculateOptimusStakingApr({
-    metrics: populationMetrics,
+  const aprMetrics = buildAprMetrics({
+    populationMetrics,
     stakingSnapshots,
     olasUsdPrice,
```

Comment:
can just pass directly MODIUS_FIXED_OLAS_PRICE_USD, no need to have a variable which is not changed anyway